pkg/prometheus/operator.go

```diff
@@ -247,3 +278,36 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 
 	return &pStatus, nil
 }
+
+// UpdateServiceMonitorStatus add the latest status in serviceMonitor selected by the Prometheus or PrometheusAgent.
+func UpdateServiceMonitorStatus(ctx context.Context, p metav1.Object, c *ConfigResourceSyncer, res TypedConfigurationResource[*monitoringv1.ServiceMonitor]) error {
+	smon := res.resource
+	conditions := res.conditions(smon.Generation)
+
+	var found bool
+	for i := range smon.Status.Bindings {
+		binding := &smon.Status.Bindings[i]
+		if binding.Namespace == p.GetNamespace() &&
+			binding.Name == p.GetName() &&
+			binding.Resource == monitoringv1.PrometheusName {
```

Comment:
should we also check the group?
```suggestion
			binding.Resource == c.gvr.Resource {
```